.changeset/long-bags-itch.md

```diff
@@ -0,0 +1,5 @@
+---
+'@keystonejs/file-adapters': major
+---
+
+Removed unnecessary required fields for S3Adapter
```

Comment:
Move `accessKeyId`, `secretAccessKey` and `region` to `s3Options` and renamed `secretAccessKey` to `s3`.

Previously:

```
const fileAdapter = new S3Adapter({
  bucket: 'bucket-name',
  accessKeyId: 'ACCESS_KEY_ID',
  secretAccessKey: 'SECRET_ACCESS_KEY',
  region: 'us-west-2',
});
```

now:

```
const fileAdapter = new S3Adapter({
  bucket: 'bucket-name',
  s3Options: {
    accessKeyId: 'ACCESS_KEY_ID',
    s3: 'SECRET_ACCESS_KEY',
    region: 'us-west-2',
  }
});
```

Reply:
A couple of things:

1/ The renaming of that property from `secretAccessKey` to `s3` was an error in the docs. Fix forthcoming.

2/ Not sure about the accuracy of your description. These weren't "moved" anywhere. It's just that we stopped accepting them on the constructor. It was always possible (as it is after this change) to supply these properties via the `s3Options` parameter.

Reply:
I don't think we need to language lawyer the changeset too much, I think the examples do must of the heavy lifting to clarify exactly what has changed and what changes a developer would need to make in order to upgrade to the new package.

If we can make that update (or something sufficiently similar) then I'm happy to give this a ✅ and get it shipped 🚀

Reply:
Cool. I've made all the changes I think need to be done. Anything else specific we're waiting for?

Reply:
Just need to update the changeset to include the info from Mike's comment then we're good to go.

Reply:
Yep. I think [this commit](https://github.com/keystonejs/keystone/pull/2967/commits/68eda7b3960f2bef224d85d037f1c9e067b622a0) takes care of that. Have I missed something?

Reply:
Yep, the changeset file (`.changeset/long-bags-itch.md`) needs to contain the text that Mike suggested in this comment thread (or something similar). This is the text that will get put into the CHANGELOG file, and is the thing that developers will refer to when upgrading this major package version, so it's important that it clearly explains the before and after of this version change so that they can successfully this package.